vignettes/Data_retrieval_via_API.Rmd

```diff
@@ -0,0 +1,507 @@
+
+### By aeid
+```{r data_by_aeid}
+# Load Level 5 MC data for an aeid
+mc5 <- tcplLoadData(lvl=5,    # data level
+               fld="aeid",    # fields to query on
+               val=704,       # values should match their corresponding 'fld'
+               type = "mc",   # data type, default, not needed
+               add.fld=FALSE) # restrict to just level 5 parameters 
+```
+
+```{r, echo = FALSE}
+printFormattedTable(mc5)
+```
+### By spid
+```{r data_by_spid}
+# Load Level 5 MC data for an aeid
+mc5 <- tcplLoadData(lvl=5,         # data level
+               fld="spid",         # fields to query on
+               val="TP0000904H05", # values should match their corresponding 'fld'
+               type = "mc",        # data type, default, not needed
+               add.fld=FALSE)      # restrict to just level 5 parameters
+```
+
+```{r, echo = FALSE}
+printFormattedTable(mc5)
+```
+### By m4id
+```{r data_by_m4id}
+# Load Level 5 MC data for an aeid
+mc5 <- tcplLoadData(lvl=5,    # data level
+               fld="m4id",    # fields to query on
+               val=1842443,   # values should match their corresponding 'fld'
+               type = "mc",   # data type, default, not needed
+               add.fld=FALSE) # restrict to just level 5 parameters
+```
+
+```{r, echo = FALSE}
+printFormattedTable(mc5)
+```
+### By DTXSID
+```{r data_by_dtxsid}
+# Load Level 5 MC data for an aeid
+mc5 <- tcplLoadData(lvl=5,           # data level
+               fld="dtxsid",         # fields to query on
+               val="DTXSID30944145", # values should match their corresponding 'fld'
+               type = "mc",          # data type, default, not needed
+               add.fld=FALSE)        # restrict to just level 5 parameters
+```
+
+```{r, echo = FALSE}
+printFormattedTable(mc5)
+```
+### By level
+
+In addition to level 5 data, levels 3, 4, 6, and 'agg' are available to pull from the API.
+
+#### mc3
+
+```{r data_level_3}
+mc3 <- tcplLoadData(lvl=3,    # data level
+               fld="m4id",    # fields to query on
+               val=1842443,   # values should match their corresponding 'fld'
+               type = "mc",   # data type, default, not needed
+               add.fld=FALSE) # restrict to just level 3 parameters  
+```
+
+```{r, echo = FALSE}
+printFormattedTable(mc3)
+```
+
+#### mc4
+
+```{r data_level_4}
+mc4 <- tcplLoadData(lvl=4,    # data level
+               fld="m4id",    # fields to query on
+               val=1842443,   # values should match their corresponding 'fld'
+               type = "mc",   # data type, default, not needed
+               add.fld=FALSE) # restrict to just level 4 parameters 
+```
+
+```{r, echo = FALSE}
+printFormattedTable(mc4)
+```
+
+#### mc6
+
+```{r data_level_6}
+mc6 <- tcplLoadData(lvl=6,    # data level
```

Comment:
It should be the "same" as the mc6 loaded from DB with add.fld = FALSE.

I added a chunk to tcplLoadData to expand level 6 out to match loading from DB -- so one row for each flag. add.fld = TRUE will still be one row for each flag, but it would also contain all of the extra fields from other levels.